src/librustc/middle/effect.rs

```diff
@@ -186,7 +186,7 @@ impl<'a, 'tcx, 'v> Visitor<'v> for EffectCheckVisitor<'a, 'tcx> {
             hir::ExprInlineAsm(..) => {
                 self.require_unsafe(expr.span, "use of inline assembly");
             }
-            hir::ExprPath(..) => {
+            hir::ExprPath(hir::QPath::Regular(..)) => {
```

Comment:
Oh, btw, the `Def` being queried is avoided where possible, because of the latter commit where the embedded/queried distinction exists. That's why some places are weird, because that way they stay simple.